Source/WebCore/html/ImageDocument.cpp

```diff
@@ -159,8 +160,8 @@ void ImageDocument::updateDuringParsing()
     if (!frame())
         return;
 
-    if (RefPtr<FragmentedSharedBuffer> buffer = loader()->mainResourceData()) {
-        if (auto* cachedImage = m_imageElement->cachedImage())
+    if (RefPtr buffer = protectedLoader()->mainResourceData()) {
+        if (auto* cachedImage = Ref { *m_imageElement }->cachedImage())
```

Comment:
The static analysis doesn't warn about this but could you please use `CachedResourceHandle` instead of `auto*`?

Reply:
Done for all four instances I found in this file.